sql/core/src/test/scala/org/apache/spark/sql/UDFSuite.scala

```diff
@@ -82,6 +83,48 @@ class UDFSuite extends QueryTest {
     assert(ctx.sql("SELECT strLenScala('test', 1)").head().getInt(0) === 5)
   }
 
+  test("UDF in a WHERE") {
+    testData.sqlContext.udf.register("oneArgFilter", (n:Int) => { n > 80 })
+
+    val result =
+      testData.sqlContext.sql("SELECT * FROM testData WHERE oneArgFilter(key)")
+    assert(result.count() === 20)
+  }
+
+  test("UDF in a HAVING") {
+    testData.sqlContext.udf.register("havingFilter", (n:Long) => { n > 5 })
+
+    val result =
+      testData.sqlContext.sql("SELECT g, SUM(v) as s FROM groupData GROUP BY g HAVING havingFilter(s)")
```

Comment:
we now prefer just having the dataset closer to the test case rather than putting them in TestData.

You can do something like this easily in each test case itself

```
val df = Seq(("red", 1), ("red", 2), ("blue", 10), ("green", 100), ("green", 200)).toDF("g", "v")
df.registerTempTable("groupData")
```